src/server/handlers.ts

```diff
@@ -1534,6 +1536,47 @@ export const addHandlers = (connection: Connection, redirectConsole = true) => {
         };
     });
 
+    function adjustLocation(
+        definition: Definition,
+        inferredLocation: Location,
+        name: string,
+    ): Location | undefined {
+        const ctx = getContext(definition.uri);
+        const status = ctx.astResolver.request(
+            definition.uri,
+            isVirtualFileSystemReady,
+        );
+        if (!status || !status.text) return;
+        const start = inferredLocation.range.start;
+        const offset = getOffset(status.text, start);
+        const symbolIndex = status.text.indexOf(name, offset);
+        if (symbolIndex === -1) return;
+        const position = getLineAndCharacter(status.text, symbolIndex);
+        return Location.create(definition.uri, {
+            start: position,
+            end: {
+                line: position.line,
+                character: position.character + name.length,
+            },
+        });
+    }
+
+    function getAdjustedLocation(definition: Definition): Location {
+        const inferredLocation = locationFromDefinition(definition);
```

Comment:
Should we just fix the function `locationFromDefinition` instead of decorating it? Why is it useful to have both?

We can just inline `getAdjustedLocation` into `locationFromDefinition`, but it will stop being trivial. Maybe the problem is that `findDefinition` should return a more precise ast node?

Or if `findDefinition` sometimes has to return a node of the whole declaration, then we may add a function that zooms from the declaration to the symbol. Or just put it into `Definition` type.